src/back/glsl/mod.rs

```diff
@@ -1771,13 +1771,20 @@ impl<'a, W: Write> Writer<'a, W> {
             // keyword which ceases all further processing in a fragment shader, it's called OpKill
             // in spir-v that's why it's called `Statement::Kill`
             Statement::Kill => writeln!(self.out, "{}discard;", level)?,
-            // Issue an execution or a memory barrier.
+            // Issue a memory barrier. Please note that to ensure visibility,
+            // OpenGL always requires a call to the `barrier()` function after a `memoryBarrier*()`
             Statement::Barrier(flags) => {
-                if flags.is_empty() {
-                    writeln!(self.out, "{}barrier();", level)?;
-                } else {
-                    writeln!(self.out, "{}groupMemoryBarrier();", level)?;
+                if flags.contains(crate::Barrier::STORAGE) {
+                    writeln!(self.out, "{}memoryBarrierBuffer();", level)?;
+                    writeln!(self.out, "{}memoryBarrierImage();", level)?;
```

Comment:
I think the current semantics of `crate::Barrier::STORAGE` doesn't include images. Trying to clarify that with the group. So we should probably remove it for now.

Reply:
I decided to add the `Image` barrier after taking a look at the naga HLSL-out writer, which produces a `DeviceMemoryBarrier()` function call. Then I looked for more information and found this: https://github.com/microsoft/DirectXShaderCompiler/blob/master/docs/SPIR-V.rst#synchronization-intrinsics
Anyway, I will remove this for now.